sympy/calculus/singularities.py

```diff
@@ -136,29 +164,38 @@ def is_strictly_decreasing(f, interval=S.Reals):
     Examples
     ========
 
-    >>> from sympy import is_decreasing
-    >>> from sympy.abc import x
+    >>> from sympy import is_strictly_decreasing
+    >>> from sympy.abc import x, y
     >>> from sympy import S, Interval, oo
-    >>> is_decreasing(1/(x**2 - 3*x), Interval.open(1.5, 3))
+    >>> is_strictly_decreasing(1/(x**2 - 3*x), Interval.open(1.5, 3))
     True
-    >>> is_decreasing(1/(x**2 - 3*x), Interval.Lopen(3, oo))
+    >>> is_strictly_decreasing(1/(x**2 - 3*x), Interval.Lopen(3, oo))
     True
-    >>> is_decreasing(1/(x**2 - 3*x), Interval.Ropen(-oo, S(3)/2))
+    >>> is_strictly_decreasing(1/(x**2 - 3*x), Interval.Ropen(-oo, S(3)/2))
     False
-    >>> is_decreasing(-x**2, Interval(-oo, 0))
+    >>> is_strictly_decreasing(-x**2, Interval(-oo, 0))
+    False
+    >>> is_strictly_decreasing(-x**2 + y, Interval(-oo, 0), x)
     False
 
     """
-    if len(f.free_symbols) > 1:
-        raise NotImplementedError('is_strictly_decreasing has not yet been '
-                                  'implemented for multivariate expressions')
-    symbol = f.free_symbols.pop()
+    f = sympify(f)
+    free_sym = f.free_symbols
+
+    if symbol is None:
```

Comment:
I see this is repeated everywhere with somewhat same error message and same structure. You should abstract this out in a private function. To be clear logic from L185 to L191 can be made a separate function.